### PR TITLE
encode delegate name patch

### PR DESCRIPTION
**Pull request by lzxm160**
all tests fail will fix later

### CODE REVIEW

File: indexprotocol/votings/stakingprotocol.go [hunk 1]
```diff
@@ -197,18 +198,34 @@ func (p *Protocol) updateAggregateStaking(tx *sql.Tx, votes *iotextypes.VoteBuck
 
 func (p *Protocol) getStakingBucketInfoByEpoch(height, epochNum uint64, delegateName string) ([]*VotingInfo, error) {
 	ret, err := p.stakingBucketTableOperator.Get(height, p.Store.GetDB(), nil)
+	if errors.Cause(err) == db.ErrNotExist {
```
Comment (lizhefeng) [inline indexprotocol/votings/stakingprotocol.go:201]: Why do we need this?

File: indexprotocol/votings/stakingprotocol.go [hunk 2]
```diff
@@ -197,18 +198,34 @@ func (p *Protocol) updateAggregateStaking(tx *sql.Tx, votes *iotextypes.VoteBuck
...
+	}
+	if err != nil {
+		return nil, errors.Wrap(err, "staking bucket table operator")
+	}
 	bucketList, ok := ret.(*iotextypes.VoteBucketList)
 	if !ok {
 		return nil, errors.Errorf("Unexpected type %s", reflect.TypeOf(ret))
 	}
 	can, err := p.stakingCandidateTableOperator.Get(height, p.Store.GetDB(), nil)
+	if errors.Cause(err) == db.ErrNotExist {
```
Comment (lizhefeng) [inline indexprotocol/votings/stakingprotocol.go:212]: Why do we need this?

File: indexprotocol/votings/stakingprotocol.go [hunk 3]
```diff
@@ -197,18 +198,34 @@ func (p *Protocol) updateAggregateStaking(tx *sql.Tx, votes *iotextypes.VoteBuck
...
+		return nil, errors.Wrap(err, "staking candidate table operator")
+	}
 	candidateList, ok := can.(*iotextypes.CandidateListV2)
 	if !ok {
 		return nil, errors.Errorf("Unexpected type %s", reflect.TypeOf(can))
 	}
 	var candidateAddress string
 	for _, cand := range candidateList.Candidates {
-		if cand.Name == delegateName {
+		encode, err := indexprotocol.EncodeDelegateName(cand.Name)
```
Comment (lizhefeng) [inline indexprotocol/votings/stakingprotocol.go:224]: encodedName, err := ...

File: indexprotocol/votings/stakingprotocol.go [hunk 4]
```diff
@@ -197,18 +198,34 @@ func (p *Protocol) updateAggregateStaking(tx *sql.Tx, votes *iotextypes.VoteBuck
 
 func (p *Protocol) getStakingBucketInfoByEpoch(height, epochNum uint64, delegateName string) ([]*VotingInfo, error) {
 	ret, err := p.stakingBucketTableOperator.Get(height, p.Store.GetDB(), nil)
+	if errors.Cause(err) == db.ErrNotExist {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, errors.Wrap(err, "staking bucket table operator")
```
Comment (lizhefeng) [inline indexprotocol/votings/stakingprotocol.go:205]: return nil, errors.Wrap(err, "failed to get staking buckets")

File: indexprotocol/votings/stakingprotocol.go [hunk 5]
```diff
@@ -197,18 +198,34 @@ func (p *Protocol) updateAggregateStaking(tx *sql.Tx, votes *iotextypes.VoteBuck
...
 	bucketList, ok := ret.(*iotextypes.VoteBucketList)
 	if !ok {
 		return nil, errors.Errorf("Unexpected type %s", reflect.TypeOf(ret))
 	}
 	can, err := p.stakingCandidateTableOperator.Get(height, p.Store.GetDB(), nil)
+	if errors.Cause(err) == db.ErrNotExist {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, errors.Wrap(err, "staking candidate table operator")
```
Comment (lizhefeng) [inline indexprotocol/votings/stakingprotocol.go:216]: return nil, errors.Wrap(err, "failed to get staking candidates")

File: indexprotocol/votings/protocol.go [hunk 6]
```diff
@@ -404,6 +404,9 @@ func (p *Protocol) calculateEthereumStaking(height uint64, tx *sql.Tx) (*types.E
 		return nil, err
 	}
 	valueOfBuckets, err := p.bucketTableOperator.Get(height, p.Store.GetDB(), tx)
+	if errors.Cause(err) == db.ErrNotExist {
```
Comment (lizhefeng) [inline indexprotocol/votings/protocol.go:407]: please remove this.

File: indexprotocol/votings/stakingprotocol.go [hunk 7]
```diff
@@ -233,9 +251,13 @@ func (p *Protocol) getStakingBucketInfoByEpoch(height, epochNum uint64, delegate
...
 				weightedVotes, _ = votingPower.Mul(votingPower, big.NewFloat(intensityRate)).Int(nil)
 			}
+			voteOwnerAddress, err := util.IoAddrToEvmAddr(vote.Owner)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to convert IoTeX address to ETH address")
+			}
 			votinginfo := &VotingInfo{
 				EpochNumber:       epochNum,
-				VoterAddress:      vote.Owner,
+				VoterAddress:      strings.TrimPrefix(voteOwnerAddress.Hex(), "0x"),
```
Comment (lzxm160) [inline indexprotocol/votings/stakingprotocol.go:260]: change to ethereum address, so frontend can show ethereum and iotex address

**Pull request merged**